src/java.desktop/share/classes/javax/swing/ImageIcon.java

```diff
@@ -176,6 +181,7 @@ public ImageIcon (String filename) {
      * Creates an ImageIcon from the specified URL. The image will
      * be preloaded by using MediaTracker to monitor the loaded state
      * of the image.
+     * Passing {@code null} URL will result in {@code NullPointerException}.
```

Comment:
In the CSR, Joe quite reasonably requested that these all become @throws clauses